docs/endpoints/config_reference.rst

```diff
@@ -0,0 +1,297 @@
+Configuration Reference
+***********************
+
+Compute endpoints are configured by a text YAML file named ``config.yaml`` in the
+endpoint directory.  (A reminder that the "endpoint directory" is a subdirectory within
+``~/.globus_compute/``, with the same name as the endpoint.)  This YAML file is a
+convenience interface to the Python configuration classes used internally by Compute
+endpoints.  Anything specified in YAML is conveyed during the endpoint startup to those
+backing classes; consequently, for a complete list of options, please see the
+:ref:`internal class documentation <config-class-doc>` at the bottom of this page.
+Meanwhile, the YAML interface is generally much easier to understand, easier to
+``diff``, and, with less required boilerplate code, easier to maintain.
+
+YAML Configuration
+==================
+
+Compute endpoints currently come in two flavors: one for processing tasks and one for
+*managing* multiple task-processing endpoints.  We discuss the former first, backed by
+|UserEndpointConfig|, as it is typically the first entry into Compute for most people.
+
+User Endpoint Configuration
+---------------------------
+
+The basic configuration has a number of options, but the only required item is
+``engine``.  Globus Compute implements three engines: ``ThreadPoolEngine``,
+``ProcessPoolEngine``, and ``GlobusComputeEngine``.  The first two are Compute endpoint
+wrappers of Python's |ThreadPoolExecutor|_ and |ProcessPoolExecutor|_.  These engines
+are most appropriate for single‑host installations (e.g., a personal workstation).  For
+scheduler‑based clusters, |GlobusComputeEngine|_, as a wrapper over Parsl's
```

Comment:
I think we should clarify that `GlobusComputeEngine` is the default engine, and chosen because its auto-scaling behavior makes it a better match for endpoints that see bursty workloads.